### DISCUSSION TITLE
Error when click on "Render init_image"

### DISCUSSION THREAD

**Discussion by edenexposito** [proceduralit/StableDiffusion_Houdini_WIP#1] [Q&A]
Hello!

I'm with H19.5.522 Py3.9
I have installed your hda and Automatic1111 with API but in "Init_image" tab when I press "Render init_image" I get this error

Error running callback:
Traceback (most recent call last):
  File "mtabasi::Object/stablediffusion::2.0/init_render", line 1, in <module>
  File "mtabasi::Object/stablediffusion::2.0, PythonModule", line 37, in initRender
  File "C:\PROGRA~1/SIDEEF~1/HOUDIN~1.522/houdini/python3.9libs\nodegraphtopui.py", line 562, in dirtyNode
    chooseDeleteOutputFiles(node)
  File "C:\PROGRA~1/SIDEEF~1/HOUDIN~1.522/houdini/python3.9libs\nodegraphtopui.py", line 439, in chooseDeleteOutputFiles
    dirty_state = pdg_node.prepareDirty()
AttributeError: 'NoneType' object has no attribute 'prepareDirty'

Do you know what could be the problem? thaaanks!

![image](https://user-images.githubusercontent.com/7686452/220265964-422330f4-cbb1-4edc-957f-87d55c829ca1.png)

**Comment by proceduralit** [comment]
Hi edenexposito!

The error says couldn't find the pdg node which is very strange and new to me. 
Your network seems very simple, so I think there is something wrong with the Houdini file, may I have your hip file?

**Comment by edenexposito** [comment]
I have tried in multiple machines with multiples Houdini version..
For some reason is needed to configure next steps and click on “generate” to “activate” this hda, after that I can render with OpenGL/mantra without issues. But I can not see the result in the viewer… as you show when we can see top/bottom ships…

**Reply by proceduralit** [reply]
Thanks for the info, I'll try to reproduce the error on my side.

Regarding checking the result in the output viewer, please ensure you've activated each work_item by clicking on the circle dot on the node.

Sometime the refresh button will help as well.

![parms_resultViewer](https://user-images.githubusercontent.com/95032849/220354718-17972010-472d-4d55-9cb9-d700b05b8619.jpg)

**Comment by howitzer99** [comment]
I am having issues with both OpenGl and Mantra, where both are causing these errors when attempting an OpenGL "Render_init_image":
Traceback (most recent call last):
  File "mtabasi::Object/stablediffusion::2.0/init_render", line 1, in <module>
  File "mtabasi::Object/stablediffusion::2.0, PythonModule", line 37, in initRender
  File "C:\PROGRA~1/SIDEEF~1/HOUDIN~1.534/houdini/python3.9libs\nodegraphtopui.py", line 562, in dirtyNode
    chooseDeleteOutputFiles(node)
  File "C:\PROGRA~1/SIDEEF~1/HOUDIN~1.534/houdini/python3.9libs\nodegraphtopui.py", line 439, in chooseDeleteOutputFiles
    dirty_state = pdg_node.prepareDirty()
AttributeError: 'NoneType' object has no attribute 'prepareDirty'

When I hit "Generate", I see the Task Grid populate, but the task just sits in an endless cooking state.  The hda never populated with the two circles you mentioned above (and seen from your video).

I have no problems with running A1111 via the API for Blender, so I know the process should work.  Hitting the "Get SD Models" button is returning my list of installed checkpoint models from my SD directory.

Specs:
Houdini 19.5.534
HDA: mtabasi.stablediffusion.2.0.hdalc
A1111 running locally with --api enabled.
Model: sd-v1-4.ckpt [fe4efff1e1]

**Reply by proceduralit** [reply]
Regarding the rendering init_image, have you checked the output folder, I mean, sometimes it throws the error but renders the image as well. 

And, for the generating process, could you please check the server respond window(cmd-like window). Does your server show the process has started and running?

**Reply by howitzer99** [reply]
No, nothing is in the init_image folder.  It's strange because when I click the button, it asks to confirm to delete the file, but there isn't an actual file there.  So seems to be running based on the input text string, and not a query for a file on disk.

I checked, and no, the webui cmd does not show any process starting.  Houdini appears to be stuck on the init generation process, and never gets past there (no population of tasks in the Tops node).  I only see the circle continually spinning, and the Task Grid showing a Cooking status that never gets past 0%.

**Reply by proceduralit** [reply]
Okay, I've updated the HDA file, could you please check this out and lemme know if it works?

Thanks

**Comment by howitzer99** [comment]
Thanks for the quick update.  Unfortunately, there is no difference other than my scene not throwing errors when loading. The HDA still will not generate an openGL init image.  I've verified my paths are ok for both the init image and the output.

Is there anything locally I can test to help solve this?

![Screenshot_10](https://user-images.githubusercontent.com/13814276/228574116-30cbea31-d8d8-44b3-99b2-69fa358cf320.jpg)

**Comment by howitzer99** [comment]
If I hit Generate, and just bypass the init image step, then I do see SD startup in the web-ui CMD and load the checkpoint model.  But afterwards, the Houdini task go into a endless Waiting state again.

![Screenshot_11](https://user-images.githubusercontent.com/13814276/228575744-614f4fb7-49d0-42af-ab61-81267f4cc372.jpg)

**Reply by proceduralit** [reply]
Without init_image it won't work correctly.
To generate the init_image manually, you can reach the render node and click the Render button.
![Screenshot 2023-03-30 095958](https://user-images.githubusercontent.com/95032849/228739625-5f732e6e-2466-4d3d-a9a8-116acfb4db9b.jpg)

**Comment by howitzer99** [comment]
Thanks, that worked!  So for now, it looks like the workflow is to punch into the HDA and manually render the "init_image.jpg" file from the ROPnet each time the scene is modified (camera moved, geo added, etc).
